src/tag.cpp

```diff
@@ -34,34 +34,18 @@ void tag_destroy() {
 
 
 HSTag::HSTag(std::string name_, Settings* settings)
-    : frame_count("frame_count", std::bind(&HSTag::computeFrameCount, this))
-    , client_count("client_count", std::bind(&HSTag::computeClientCount, this))
-    , curframe_windex("curframe_windex",
+    : index(this, "index", 0)
+    , floating(this, "floating", false, [](bool){return "";})
+    , name(this, "name", name_, &HSTag::validateNewName)
+    , frame_count(this, "frame_count", &HSTag::computeFrameCount)
+    , client_count(this, "client_count", &HSTag::computeClientCount)
+    , curframe_windex(this, "curframe_windex",
```

Comment:
I don't see anything about this lambda variant in itself, but I'm wondering:
Maybe it would be better to reduce the number of ctor variants in `DynAttribute_` by dropping this one? In this particular case, `HSTag` would need to provide additional getters that are basically just these one-liners.

It's clearly just a trade-off, but I feel that additional getters would "cost less" in terms of complexity than this ctor variant does at *both* its point of usage (i.e., right here) *and* within `DynAttribute_`.

Reply:
I just realized that you've actually already been doing it exactly like that before, just with `BIND_THIS`. So you've probably got your reasons... feel free to ignore me here.

Reply:
I'm fine with introducing additional getters.

However in settings.cpp, e.g. for `border_width`, we will still need the `DynAttribute_` constructor that has a lambda as a getter and as a setter because in the settings, just a member pointer is not enough.

Reply:
I guess it's true, but I don't fully understand why. Is it because there are a lot of attributes that are very similar and you don't want to turn this single `getIntAttr()` helper into `n` almost identical getters, right?

Reply:
Right. I want to re-use `getIntAttr()` for every int typed setting. But this isn't possible anymore if we drop the constructor accepting the lambdas. This issue is also connected with the issue that I need to use C++-style member function pointers and can't use lambdas in the templated constructor of `DynAttribute_`.

Reply:
Hmmm. I'm wondering if there is a good workaround. But that's not really relevant here. So I see why we should keep the lambda constructor variant. At the same time appreciate that you're open to adding simple getters instead of using lambdas wherever possible.

I feel that in the concrete case of HSTag, readability of the initializer list would be improved by sticking to the non-lambda variant.

Reply:
My conclusion is that we need both but should use the non-lambda whereever possible.